app/features/signup/singup-form.tsx

```diff
@@ -0,0 +1,131 @@
+
+const validateEmail = buildEmailValidator('Invalid email');
+
+export function SignupForm() {
+  const { signUp } = useOpenSecret();
+  const { toast } = useToast();
+  const {
+    register,
+    handleSubmit,
+    watch,
+    formState: { errors, isSubmitting },
+  } = useForm<FormValues>();
+  const password = watch('password');
+
+  const onSubmit: SubmitHandler<FormValues> = async (data) => {
+    try {
+      await signUp(data.email, data.password, '');
+    } catch {
+      toast({
+        variant: 'destructive',
+        title: 'Error! Signup failed',
+        description: 'Please try again later or contact support',
+      });
+    }
+  };
+
+  return (
+    <Card className="mx-auto max-w-sm">
+      <CardHeader>
+        <CardTitle className="text-2xl">Signup</CardTitle>
+        <CardDescription>
+          Enter your email & password below to setup a wallet
+        </CardDescription>
+      </CardHeader>
+      <CardContent>
+        <form
+          className="grid gap-4"
+          onSubmit={handleSubmit(onSubmit)}
+          noValidate
+        >
+          <div className="grid gap-2">
+            <Label htmlFor="email">Email</Label>
+            <Input
+              id="email"
+              type="email"
+              placeholder="satoshi@nakamoto.com"
+              aria-invalid={errors.email ? 'true' : 'false'}
+              {...register('email', {
+                required: 'Email is required',
+                validate: validateEmail,
+              })}
+            />
+            {errors.email && (
+              <span role="alert" className="text-red-500 text-sm">
+                {errors.email.message}
+              </span>
+            )}
+          </div>
+          <div className="grid gap-2">
+            <Label htmlFor="password">Password</Label>
+            <Input
+              id="password"
+              type="password"
+              aria-invalid={errors.password ? 'true' : 'false'}
+              {...register('password', {
+                required: 'Password is required',
+                minLength: {
+                  value: 8,
+                  message: 'Password must have at least 8 characters',
+                },
+              })}
+            />
+            {errors.password && (
+              <span role="alert" className="text-red-500 text-sm">
+                {errors.password.message}
+              </span>
+            )}
+          </div>
+          <div className="grid gap-2">
+            <Label htmlFor="password">Confirm Password</Label>
+            <Input
+              id="confirmPassword"
+              type="password"
+              aria-invalid={errors.confirmPassword ? 'true' : 'false'}
+              {...register('confirmPassword', {
+                required: 'Password confirmation is required',
+                validate: (value) =>
+                  value === password || 'Passwords do not match',
```

Comment:
Any reason to use `watch('password')` instead of:
```ts
                  value === getValues('password') || 'Passwords do not match',
```

Reply:
not really. I just found some example with watch and asked gpt as well. we can change it to getValues

Reply:
actually seems like even this is possible:
```
validate: (value, values) =>
                  value === values.password || 'Passwords do not match'
```